Dockerfile

```diff
@@ -1,3 +1,35 @@
+# This stage extracts all the pom.xml files.
+# It'll get rebuilt with any source change, but that's OK.
+FROM openjdk:11-jdk AS poms
+WORKDIR /usr/src
+COPY . .
+# Wipe any files not called pom.xml or *.jar
+RUN find . -type f -and \! -name pom.xml -and \! -name '*.jar' -delete
+# Clear up any (now) empty diretories
+RUN find . -type d -empty -delete
+# Just for debug:
+RUN find
+
+# Now we build:
+FROM openjdk:11-jdk AS build
+# Set up Maven. No need to clean caches, this doesn't end up in the runtime container
+RUN apt-get update -qq
+RUN apt-get install -qqy maven
```

Comment:
Agree with Guus' point - is there a good reason not to use the `./mvnw` wrapper used instead? I don't think it'd affect caching, right?